core/include/optee_rpc_cmd.h

```diff
@@ -148,6 +148,18 @@
  */
 #define OPTEE_RPC_CMD_BENCH_REG		20
 
+/*
+ * Transfer data in/out via I2C
+ *
+ * [in]     value[0].a	    Transfer mode (read/write).
+ * [in]     value[0].b	    The I2C bus.
+ * [in]     value[0].c	    The I2C chip to transfer from/to.
+ *
+ * [in/out] memref[1]	    Buffer used to transfer the data.
+ * [in]     value[2].a	    Number of bytes transferred by the kernel driver.
```

Comment:
Is value[2] needed? `memref[1].size` already provides the input and output sizes.

Reply:
yes it is needed: it reports the actual number of bytes processed by the linux driver.

Reply:
ok.

Note that output memref[1].size will also provide the information for read case.
As for the input case (write), you don't really care how many bytes were written, you expect TEE_SUCCESS <=> all bytes written.

Reply:
Discussed from https://github.com/OP-TEE/optee_os/pull/3905#issuecomment-646825893 and https://github.com/etienne-lms.

By the way, replace `kernel` by `non-secure world`.